### PR TITLE
Messages polling fixed

### PR DESCRIPTION
**Pull request by miha-stopar**
Inbox.poll function was executing callback before parseRawMessages was finished. Now parseRawMessages uses async for decrypting the messages and executes the callback (passed from poll) when all messages are decrypted.

### CODE REVIEW

**Comment by daviddahl** [comment]
Cool. I am sorry I did not see this PR until now. Reviewing now.

File: client/src/inbox.js [hunk 1]
```diff
@@ -149,22 +146,20 @@ Inbox.prototype.clear = function (callback) {
...
-    }
-
-    var message = new crypton.Message(this.session, rawMessage);
-    message.decrypt(function (err) {
-      // XXXddahl: fix this, check for error
-      that.messages[message.messageId] = message;
-    });
-  }
+  async.each(this.rawMessages, function (rawMessage, callback) {
+ 	var message = new crypton.Message(that.session, rawMessage);
```
Comment (daviddahl) [inline client/src/inbox.js]: nit: 2 space indent, please (throughout all JS)

File: test/test/inbox.js [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+      crypton.authorize('alice', 'pass', function (err, session) {
+        if (err) throw err;
+        aliceSession = session;
+        done();
+      });
+
+      // this stringify function is to avoid the following error in mocha: 
+      // Uncaught TypeError: JSON.stringify cannot serialize cyclic structures
+      JSON.stringify = function(obj) {
```
Comment (daviddahl) [inline test/test/inbox.js]: Nice!

**Comment by daviddahl** [comment]
LGTM. Just the indention issues and we can merge it.

**Comment by miha-stopar** [comment]
Indentation fixed.

**Pull request merged**